demo/objectFusionTracksLoader.js

```diff
@@ -1,215 +1,386 @@
+// import { Flatbuffer } from "../schemas/GroundTruth_generated.js";
+// import { Flatbuffer } from "http://localhost:1234/schemas/GroundTruth_generated.js";
+import { updateLoadingBar, incrementLoadingBarTotal } from "../common/overlay.js";
+import { getFbFileInfo, removeFileExtension } from "./loaderUtilities.js";
 
 
-let objectFusionTracksFiles = null;
+// sets local variable and returns so # files can be counted
+let trackFiles = null;
 export const objectFusionTracksDownloads = async (datasetFiles) => {
-  objectFusionTracksFiles = await getFbFileInfo(datasetFiles,
-                                       "sf_bounding_boxes.fb",
-                                       "2_Truth",
-                                       "GroundTruth_generated.js", // 5_Schemas
-                                       "../data/sf_bounding_boxes.fb",
-                                       "../schemas/GroundTruth_generated.js");
-  return objectFusionTracksFiles;
+  trackFiles = await getFbFileInfo(datasetFiles,
+                                   "object_fusion_tracks.fb",
+                                   "3_Assessments",
+                                   "GroundTruth_generated.js", // 5_Schemas
+                                   "../data/object_fusion_tracks.fb",
+                                   "../schemas/GroundTruth_generated.js");
+  return trackFiles;
 }
 
-async function loadDetections(file, shaderMaterial, animationEngine) {
+export async function loadTracks(s3, bucket, name, trackFileName, shaderMaterial, animationEngine, callback) {
+  const tstart = performance.now();
+  if (!trackFiles) {
+    console.log("No track files present")
+    return
+  }
 
-  if (!objectFusionTracksFiles) {
-    console.log("No detection files present")
-    return null
-  } else {
-    // prepare for progress tracking (currently only triggered on button click)
-    resetProgressBars(2); // have to download & process/load detections
+  if (trackFileName) {
+    trackFiles.objectName = `${name}/3_Assessments/${trackFileName}`;
   }
+  console.log("tracks", trackFileName);
+  console.log("tracks load", trackFiles.objectName);
+  if (s3 && bucket && name) {
+    (async () => {
+      const schemaUrl = s3.getSignedUrl('getObject', {
+        Bucket: bucket,
+        Key: trackFiles.schemaFile
+      });
+      const request = await s3.getObject({Bucket: bucket,
+        Key: trackFiles.objectName},
+        async (err, data) => {
+          if (err) {
+            console.error("Error getting tracks file", err, err.stack);
+          } else {
+            const FlatbufferModule = await import(schemaUrl);
+            const trackGeometries = await parseTracks(data.Body, shaderMaterial, FlatbufferModule, animationEngine);
+            await callback(trackGeometries, );
+          }
+          incrementLoadingBarTotal("tracks loaded")
+        });
+      request.on("httpDownloadProgress", async (e) => {
+        await updateLoadingBar(e.loaded/e.total * 100);
+      });
```

Comment:
You don't need `async` or `await` here. The progress thing gets called, and no one is waiting for the answer.